ludwig/utils/triton_utils.py

```diff
@@ -49,142 +95,538 @@ def forward(self, {input_signature}):
-        input_spec=_get_input_spec(model), output_spec=_get_output_spec(model), instance_spec=_get_instance_spec()
-    )
-    return config
+    module: Union[_InferencePreprocessor, _InferencePredictor, _InferencePostprocessor]
+    input_features: List[TritonConfigFeature]
+    output_features: List[TritonConfigFeature]
+    inference_stage: str
+
+    def _get_dict_type_hint(self) -> str:
+        return {
+            PREPROCESSOR: "TorchscriptPreprocessingInput",
+            PREDICTOR: "torch.Tensor",
+            POSTPROCESSOR: "torch.Tensor",
+        }[self.inference_stage]
+
+    def _get_input_signature(self, triton_features: List[TritonConfigFeature]) -> str:
+        elems = [
+            f"{feature.wrapper_signature_name}: {feature._get_wrapper_signature_type()}" for feature in triton_features
+        ]
+        return ", ".join(elems)
+
+    def _get_input_dict(self, triton_features: List[TritonConfigFeature]) -> str:
+        elems = [f'"{feature.name}": {feature.wrapper_signature_name}' for feature in triton_features]
+        return "{" + ", ".join(elems) + "}"
+
+    def _get_output_tuple(self, triton_features: List[TritonConfigFeature]) -> str:
+        elems = [f'results["{feature.name}"]' for feature in triton_features]
+        return "(" + ", ".join(elems) + ",)"
+
+    def generate_inference_module_wrapper(self) -> str:
+        """Generate the class wrapper around an inference module."""
+        return INFERENCE_MODULE_TEMPLATE.format(
+            input_signature=self._get_input_signature(self.input_features),
+            input_type=self._get_dict_type_hint(),
+            input_dict=self._get_input_dict(self.input_features),
+            output_tuple=self._get_output_tuple(self.output_features),
+        )
+
+    def generate_scripted_module(self):
+        """Generate the scripted module from the wrapper class."""
+        wrapper_definition = self.generate_inference_module_wrapper()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            ts_path = os.path.join(tmpdir, "generated.py")
+            with open(ts_path, "w") as f:
+                f.write(wrapper_definition)
+
+            spec = importlib.util.spec_from_file_location("generated.ts", ts_path)
+            gen_ts = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(gen_ts)
+
+            gen_module = gen_ts.GeneratedInferenceModule(self.module)
+            scripted_module = torch.jit.script(gen_module)
+        return scripted_module
 
 
 def export_triton(
-    model: LudwigModel, output_path: str, model_name: str = "ludwig_model", model_version: Union[int, str] = 1
-) -> Tuple[str, str]:
+    model: LudwigModel,
+    data_example: pd.DataFrame,
+    predictor_max_batch_size: int = 64,
+    max_queue_delay_microseconds: int = 100,
+    output_path: str = "model_repository",
+    model_name: str = "ludwig_model",
+    model_version: Union[int, str] = 1,
+    device: str = "cpu",
+    device_count: int = 1,
```

Comment:
In order to better reflect the ability to target a different number of cpu and gpu instances, I suggest we change the configuration so its close to ray [init](https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-init) eg:
* `num_cpus: Optional[int]=1`,
* `num_gpus: Optional[int]=0`,
* `max_batch_size: Optional[int]=64`,
* `max_queue_delay_microseconds: Optional[int]=100`
From here we can use the `num_gpus>0` to determine if we target the gpu device for the predictor